### PR TITLE
docs: document W3C tracecontext header

### PR DESCRIPTION
**Pull request by jahtalab**
Part of #477

* add backend agents matrix

### CODE REVIEW

File: docs/distributed-tracing-guide.asciidoc [hunk 1]
```diff
@@ -55,17 +55,38 @@ And should respond to it with these headers:
...
 NOTE: To make sure all components in a distributed trace are included,
 the sampling rate of backend agents might be affected by the sampling rate of the RUM agent.
 
+
+[float]
+[[backend-agents-compatibility]]
+=== Backend agents compatibility
+Since version 5.0, the RUM agent supports the official W3C tracecontext `traceparent` header,
+instead of the previously used `elastic-apm-traceparent` header.
+In this table you can find which versions of our backend agents also supports the official W3C
```
Comment (vigneshshanmugam) [inline docs/distributed-tracing-guide.asciidoc]: I feel this a bit hard to follow. What about something like this? 
`In the below compatibility table, you can find the versions of our backend
agents that can be used along with the RUM agent(>5.0) to propagate the trace via the
official tracecontext spec.`

File: docs/distributed-tracing-guide.asciidoc [hunk 2]
```diff
@@ -55,17 +55,38 @@ And should respond to it with these headers:
...
 the sampling rate of backend agents might be affected by the sampling rate of the RUM agent.
 
+
+[float]
+[[backend-agents-compatibility]]
+=== Backend agents compatibility
+Since version 5.0, the RUM agent supports the official W3C tracecontext `traceparent` header,
+instead of the previously used `elastic-apm-traceparent` header.
+In this table you can find which versions of our backend agents also supports the official W3C 
+tracecontext header and therefore can be used with the RUM agent version >=5.0 for distributed tracing:
```
Comment (bmorelli25) [inline docs/distributed-tracing-guide.asciidoc]:
```suggestion
[[backend-agent-compatibility]]
=== Backend Agent compatibility

Starting in version 5.0, the RUM Agent supports the official W3C tracecontext `traceparent` header,
instead of the previously used `elastic-apm-traceparent` header.
Use the table below to determine which versions of our backend agents also support the official W3C
tracecontext headers. Compatible agents use the official tracecontext spec to propagate traces and can
therefor be used with the RUM Agent version >=5.0 for distributed tracing.
```

**Comment by bmorelli25** [comment]
I opened https://github.com/elastic/apm-agent-rum-js/issues/671 to create an upgrade guide.

File: docs/distributed-tracing-guide.asciidoc (hunk 1 above)
Comment (jahtalab) [reply docs/distributed-tracing-guide.asciidoc]: I've applied Brandon's suggestion which improves this part.
